agent/eni/netwrapper/netwrapper_windows.go

```diff
@@ -0,0 +1,80 @@
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package netwrapper
+
+import (
+	"net"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+// NetWrapper interface is created to abstract Golang's net package from its usage in the watcher
+// Also, this enables us to mock this interface for unit tests
+type NetWrapper interface {
+	FindInterfaceByIndex(int) (*net.Interface, error)
+	GetAllNetworkInterfaces() ([]net.Interface, error)
+	GetAdapterAddresses() ([]*windows.IpAdapterAddresses, error)
```

Comment:
Why do you need to return data using `*windows.IpAdapterAddresses`? 
Can they be converted to a more user-friendly format?

Also, why has this now become platform specific? Can you please keep it agnostic??

Reply:
So this package was introduced in task networking feature to abstract Go's net package. `GetAdapterAddresses` method is copied from the Go's Windows package and the data obtained in this Win32 call can be parsed by the caller as needed.
So, for example, in Go lang's net package, Interface object is created using data returned from this method. Other data is discarded.
Similarly, we are using DNS part of the data returned by this method.

Since, this interface itself was for wrapping Windows related methods, it seems to be the apt place for it to go.